### PR TITLE
✨ Add `oracledb` support

### LINKED ISSUE #24: Add support for new oracledb package
It looks like Oracle is changing their driver package to be (mostly) pure Python, and is changing some of the interfaces.

Here's a link to some documentation that shows all of the differences:
https://python-oracledb.readthedocs.io/en/latest/user_guide/appendix_c.html

### PR DESCRIPTION
**Pull request by troyswanson**
This closes #24

### CODE REVIEW

File: tests/test_oracle/test_oracledb.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+def test_using(server, database_name):
+    with using(
+        oracledb.connect(password="pydapper", user="pydapper", dsn=f"{server}:1522/{database_name}")
+    ) as commands:
+        assert isinstance(commands, OracledbCommands)
+
+
+@pytest.mark.parametrize("driver", ["oracle", "oracle+oracledb"])
+def test_connect(driver, database_name, server):
```
Comment (zschumacher) [inline tests/test_oracle/test_oracledb.py:35]: you can remove the parameterize, I think I'm going to leave cx oracle as the oracle default for now until I can confirm that doesn't affect others

**Comment by zschumacher** [comment]
tests are failing - just need to add the new package to the poetry install in the test GHA

File: tests/test_oracle/test_oracledb.py (hunk 1 above)
Comment (troyswanson) [reply tests/test_oracle/test_oracledb.py:35]: Do you mean I should delete this test entirely?

**Comment by troyswanson** [comment]
I think we need to resolve #26 before the checks will pass again.

File: tests/test_oracle/test_oracledb.py (hunk 1 above)
Comment (troyswanson) [reply tests/test_oracle/test_oracledb.py:36]: I just changed it to test the usage of the `oracle+oracledb` DSN. Let me know if that works!

Comment (zschumacher) [reply tests/test_oracle/test_oracledb.py:36]: thats perf, ty

**Comment by zschumacher** [comment]
> I think we need to resolve #26 before the checks will pass again.

should be good now, just bumped black

**Comment by troyswanson** [comment]
OK, just force pushed some cleaned up commits on top of the latest main branch.

**Comment by zschumacher** [comment]
Everything looks good to me.  I'm still lacking a docs test unfortunately, would you mind running `make docs` and just sanity checking?

**Comment by troyswanson** [comment]
> Everything looks good to me. I'm still lacking a docs test unfortunately, would you mind running `make docs` and just sanity checking?

Yeah I just did and I found an issue. Fixed in 88b75f0 -- I think we're good to go!

**Review by zschumacher — APPROVED** [review]
Looks good, thanks for doing this 👍

**Pull request merged**